pkg/ccl/storageccl/import.go

```diff
@@ -265,50 +276,33 @@ func evalImport(ctx context.Context, cArgs batcheval.CommandArgs) (*roachpb.Impo
 			continue
 		}
 
+		// Check if we need to flush current batch *before* adding the next k/v --
+		// the batcher may want to flush the keys it already has, either because it
+		// is full or because it wants this key in a separate batch due to splits.
```

Comment:
"or because it wants this key in a separate batch due to splits"

where does this happen? how did this work correctly before?

Reply:
it *doesn't* work correctly: the implementation currently of `shouldFlush` doesn't even look at the passed key, but it will Soon™. I just pulled this refactor out of the bigger CL that adds that split and retry.